app/src/main/java/org/walleth/data/TokenPrimer.kt

```diff
@@ -39,16 +39,18 @@ fun initTokens(settings: Settings, assets: AssetManager, appDatabase: AppDatabas
 
                     val newTokens = (0 until jsonArray.length()).map { jsonArray.get(it) as JSONObject }.map {
                         val address = it.getString("address")
+                        val symbol = it.getString("symbol")
                         Token(
-                                symbol = it.getString("symbol"),
+                                symbol = symbol,
                                 name = it.getString("name"),
                                 decimals = Integer.parseInt(it.getString("decimals")),
                                 address = Address(address),
                                 starred = false,
                                 showInList = true,
                                 fromUser = false,
                                 chain = chain,
-                                order = mapToOrder(address)
+                                order = mapToOrder(address),
+                                requiresBalance = "BeerCoin" != symbol
```

Comment:
How to generalize this?

Reply:
I would suggest the following:
 this needs an interface that implements mint() - because in the end you mint one IOU before transfer() - then you expose your interfaces via  ERC-165
 WallETH can then detect the mint() function via 165 and offer a button. Needs some rewrite of the BeerCoin contract - perhaps rename in this case als to something more healthy ;-)